### PR TITLE
fix handling of file urls on windows

### PR DESCRIPTION
**Pull request by MichaelBeier**
Hi,

i encountered an issue with the handling of file urls on windows. Characters inside the url may be encoded. This causes the comparison inside includesFile in GraphQLProjectConfig to fail.

This problem can be solved by passing the file uri to fileURLToPath (https://nodejs.org/dist/latest-v12.x/docs/api/url.html#url_url_fileurltopath_url) method of nodes url module. 
Since this function is only available from 10.12 onwards, i implemented a minimalistic version of it inside the includesFile method.

### CODE REVIEW

**Comment by ardatan** [comment]
@MichaelBeier Since we dropped Node 8 support, do you think we can use the builtin function you mentioned instead?

**Comment by MichaelBeier** [comment]
Yes, it should work the same/better.

You're dropping node 8 for version 3, right? I was hoping to get a patch for version 2, since that's the version being used by https://github.com/prisma-labs/vscode-graphql, and i'm not sure about their timeline to upgrade to version 3.

**Comment by kamilkisiela** [comment]
@ardatan when do we dropped Node 8? Could you point out the commit?

File: src/GraphQLProjectConfig.ts [hunk 1]
```diff
@@ -40,9 +43,19 @@ export class GraphQLProjectConfig {
...
 
   includesFile(fileUri: string): boolean {
-    const filePath = fileUri.startsWith('file://')
-      ? fileUri.substr(7)
-      : fileUri;
+    let filePath = fileUri;
+
+    if (fileUri.startsWith('file://')) {
+      if (isWindows) {
+        filePath = filePath.substr(8);
```
Comment (kamilkisiela) [inline src/GraphQLProjectConfig.ts]: Why do we use `8` here? What if there's `file://foo.js`

Comment (dotansimha) [reply src/GraphQLProjectConfig.ts:50]: It just removes the `file://`, it should work the same

**Comment by ardatan** [comment]
Thanks @MichaelBeier !

**Comment by MichaelBeier** [comment]
Oh, i forgot about this.

Thanks for finishing it.

**Pull request merged**